### DISCUSSION TITLE
Is there a way to get the values of tags from the Graphics State Parameter Dictionary using PyMuPDF, especially in the case of drawings?

### DISCUSSION THREAD

**Discussion by SirishaGorasa** [pymupdf/PyMuPDF#2356] [Looking for help] [answered]
When we extract Graphics using page.get_drawings() or page.get_cdrawings() methods of PyMuPDF, we get certain information about wfh graphics object. For every drawing, there would be a Graphics State Parameter Dictionary (GS) in the internal structure. A Sample Graphics State Dictionary attached below:

![image](https://user-images.githubusercontent.com/47634813/232482341-34c53c85-035a-4e83-bd7a-804fa2c410df.jpeg)

The above screenshot is the information corresponding to a graphics state.

I want to know if there is a way to extract these properties for a path object from the internal structure of a PDF using PyMuPDF.

**Comment by JorjMcKie** [comment, marked as answer]
No, sorry this is not possible like this. The reason being that `get_drawings()` (like also `get_text()`) is **_not_** PDF-specific, but works for all MuPDF-supported document types like EPUB or XPS.
And they do not have the kind of things you are mentioning.

With the new version 1.22.0 there are more data available:
* New key "layer":the name of the optional content group or `None`.
* If using `page.get_drawings(extended=True)`, clipping paths are returned, as well as so-called group paths. The remaining "standard" paths (stroke, fill, fill-stroke) then have a new "level" key (integer) which indicates the ownership to some preceding clip or group. Group paths contain information like blend mode. Fairly complex stuff, so please make sure to read the documentation.

**Reply by SirishaGorasa** [reply]
Could you once let me know if the newer version of PyMuPDF 1.22.0 give the values to keys like Blend mode (BM), Stroke Adjustment (SA), Overprint non-stroking(op), Overprint stroking(OP), AlphaisShape(AIS), /SMask and /OPM??? If possible, Could you please provide me the new version's documentation URL?

**Reply by JorjMcKie** [reply]
Here is a documentation of the changes. They will become part of the official documentation shortly.
[get-drawings.pdf](https://github.com/pymupdf/PyMuPDF/files/11251685/get-drawings.pdf)

**Reply by SirishaGorasa** [reply]
Thanks for this 😊
Thanks a lot for the quick response.
